applications/FluidDynamicsApplication/custom_strategies/strategies/compressible_navier_stokes_explicit_solving_strategy_runge_kutta_4.h

```diff
@@ -0,0 +1,834 @@
+                if (eigen_val_mat(d,d) > shear_spect_norm) {
+                    shear_spect_norm = eigen_val_mat(d,d);
+                }
+            }
+            const double isentropic_max_vel = std::sqrt(v_norm_pow + (2.0 / (gamma - 1.0)) *  std::pow(c_ref, 2)); // TODO: THIS ISN'T c_ref ACCORDING TO PERAIRE 
+
+            const double s_mu_0 = 1.0;
+            const double s_mu_max = 2.0;
+            const double s_mu = h_ref * shear_spect_norm / isentropic_max_vel / k;
+            // const double s_mu_hat = LimitingFunction(s_mu, s_mu_0, s_mu_max);
+            const double s_mu_hat = SmoothedLimitingFunction(s_mu, s_mu_0, s_mu_max);
+            it_elem->GetValue(SHEAR_SENSOR) = s_mu_hat;
+
+            // Calculate artificial magnitudes
+            const double ref_mom_norm = midpoint_rho * std::sqrt(v_norm_pow + std::pow(c_ref,2));
+
+            // Calculate elemental artificial bulk viscosity
+            const double k_beta = 1.5;
+            const double elem_b_star =  (k_beta * h_beta / k) * ref_mom_norm * s_beta_hat; 
+            it_elem->GetValue(ARTIFICIAL_BULK_VISCOSITY) = elem_b_star;
+
+            // Calculate elemental artificial conductivity (dilatancy)
+            const double Pr_beta_min = 0.9;
+            const double alpha_pr_beta = 2.0;
+            const double Mach_threshold = 3.0;
+            const double Mach = norm_2(midpoint_v) / c_ref;
+            const double Pr_beta = Pr_beta_min * (1.0 + std::exp(-2.0 * alpha_pr_beta * (Mach - Mach_threshold)));
+            const double elem_k1_star = (gamma * c_v / Pr_beta) * elem_b_star;
+
+            // Calculate elemental artificial conductivity (thermal sensor)
+            const double k_kappa = 1.0;
+            const double elem_k2_star = (gamma * c_v) * (k_kappa * h_kappa / k) * ref_mom_norm * s_kappa_hat;
+            it_elem->GetValue(ARTIFICIAL_CONDUCTIVITY) = elem_k1_star + elem_k2_star;
+
+            // Calculate elemental artificial dynamic viscosity
+            const double k_mu = 1.0;
+            const double elem_mu_star = (k_mu * h_mu / k) * ref_mom_norm * s_mu_hat;
+            it_elem->GetValue(ARTIFICIAL_DYNAMIC_VISCOSITY) = elem_mu_star;
+
+            // Project the shock capturing magnitudes to the nodes
+            const double geom_domain_size = r_geom.DomainSize();
+            const double aux_weight = geom_domain_size / static_cast<double>(n_nodes);
+            for (unsigned int i_node = 0; i_node < n_nodes; ++i_node) {
+                auto& r_node = r_geom[i_node];
+#pragma omp atomic
+                r_node.GetValue(ARTIFICIAL_BULK_VISCOSITY) += aux_weight * elem_b_star;
+#pragma omp atomic
+                r_node.GetValue(ARTIFICIAL_CONDUCTIVITY) += aux_weight * (elem_k1_star + elem_k2_star);
+#pragma omp atomic
+                r_node.GetValue(ARTIFICIAL_DYNAMIC_VISCOSITY) += aux_weight * elem_mu_star;
+#pragma omp atomic
+                r_node.GetValue(NODAL_AREA) += aux_weight;
```

Comment:
i would argue that this function is WAY TOO LONG

Reply:
I think it is more than way too long :smile:. Considering that this work and also that we will move this to an independent utility, I'd like to keep it as it is for the moment.